addons/sale/controllers/portal.py

```diff
@@ -284,13 +283,8 @@ def portal_quote_accept(self, order_id, access_token=None, name=None, signature=
 
         pdf = request.env['ir.actions.report'].sudo()._render_qweb_pdf('sale.action_report_saleorder', [order_sudo.id])[0]
 
-        order_sudo.message_post(
+        order_sudo.with_user(SUPERUSER_ID).message_post(
             attachments=[('%s.pdf' % order_sudo.name, pdf)],
-            author_id=(
-                order_sudo.partner_id.id
-                if request.env.user._is_public()
-                else request.env.user.partner_id.id
-            ),
```

Comment:
I believe it might be interesting to know who signed the quote, there is no guarantee that the provided name is the user name :eyes: